### ISSUE TITLE
TypeScript has not added regression tests for the regressions

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#30144]
Regression tests for #30133 which made a lot of regressions are not added. Unfortunately the blind maintainers couldn't find my reports but at least you must add the tests written in #30118 #30049 #30047 to recover your blunder.

**Comment by falsandtru** [comment]
Or to add regression tests for the large regressions you missed is also unactionable?

**Comment by jack-williams** [comment]
Here you go #30146. It's up to the maintainers to decide if they want to merge this. Note: it's currently 3am in Seattle.

**Comment by falsandtru** [comment]
Nice but why you excluded #30049? And can you include my actual example code? With TypeScript, test cases usually include the reported code for reproducing in bug fixing.

**Comment by jack-williams** [comment]
I excluded #30049 because I did not have chance to look into that; I don't want to add a test case I don't understand. I think that all the issues reduce down to the same problem, so I'm also reluctant to add more noise to the tests. Someone on the team has been assigned to look into each of these issues, so they are not 'blind'.

I used stripped down code examples that are easy to understand and directly target the issues. I don't have the time right now to change them. You are welcome to submit a fix and I will close my PR.

**Comment by RyanCavanaugh** [comment]
@falsandtru your attitude here is getting a bit rough. We're all trying our best here and the constant shade throwing in the past few reports is really not helpful.

If you can't report bugs here without name-calling, we'll have to ask you to not participate at all.

**Comment by falsandtru** [comment]
So you will miss the next large regression again until you find it yourself? It is foolish. You must learn from this mistake. Or you don't learn anything? Cool!

**Comment by RyanCavanaugh** [comment]
To be clear: I'm just asking you to not insult people.

**Comment by falsandtru** [comment]
How did you want to be explained your mistake? I don't know English enough to do it.

**Comment by DanielRosenwasser** [comment]
```diff
- Unfortunately the blind maintainers couldn't find my reports but at least you must add the tests written in #30118 #30049 #30047 to recover your blunder.
+ Unfortunately the maintainers didn't see my reports in #30118 #30049 #30047.
+ This has been extremely frustrating for me.
```

**Comment by falsandtru** [comment]
Unfortunately, I mean **how to realize you the impact of your mistake in the first reports.** You misread again for yourself, awesome work.

**Comment by falsandtru** [comment]
Anyway your answer is not smart, right?

> This has been extremely frustrating for me.

**Comment by DanielRosenwasser** [comment]
Again, you're using an abrasive tone. If I have misunderstood something for whatever reason, I'm sorry. If I've misrepresented how you feel about the situation, I'm sorry about that too. But your responses

> Unfortunately, I mean **how to realize you the impact of your mistake in the first reports.** You misread again for yourself, awesome work.

> Anyway your answer is not smart, right?

are not constructive and are making it difficult to collaborate.

**Comment by falsandtru** [comment]
I mean how can you prevent the next oversight of the terrible regression such as #27697? If you missed the next terrible regression again, I have to notify it again noisily because it is a terrible regression.

**Comment by RyanCavanaugh** [comment]
> I have to notify it again noisily because it is a terrible regression.

You don't have to do it "noisily". We don't prioritize bug reports based on what kinds of insults are included with them. Thousands of people have successfully reported issues without insulting anyone. No one *needs* to call anyone names to get a bug report noticed here.

If you have determined that you're not able to report an issue without being abrasive or abusive about it, then we ask that you not report issues or comment on them. Other people who are able to report issues in a respectful manner will fill in the gap.

**Comment by weswigham** [comment]
> So you will miss the next large regression again until you find it yourself? It is foolish. You must learn from this mistake. Or you don't learn anything? Cool!

FYI, we didn't miss this, either - we picked this up via DT within a week or so (and have since improved our DT testing infra so we can more quickly know when we break things on DT (we should know within a day reliably now) - though it was mostly slow because we all live in Seattle and were coming off being snowed in/OOF the prior week, lol). We just spent awhile attacking the issue from multiple angles because we didn't _want_ to revert #27697 if we didn't need to, and the interplay in many of the breaks with one of the other PRs merged at the same time made determining the best "fix" a longer task (and in the end we merged a PR with a modification on the other PR _and_ a revert of #27697). Relax - #27697 was only in `master` for two weeks. 

Anyways, I've merged #30146, since more tests are always gucci, so I think we're done here~
